File: Template/Persistence/src/main/java/Repository/IUserRepository.java
package Repository;

import Model.User;

import java.util.Optional;

public interface IUserRepository extends IRepository<Long, User> {
    /**
     * Finds a user by the passed credentials.
     *
     * @param username Username of the user.
     * @param password Password of the user.
     * @return An {@code Optional} containing the user if the user has been found.
     */
    Optional<User> findUserByCredentials(String username, String password) throws RepositoryException;
}

File: MotorcycleContest/MotorcycleContest/Repository/IUserRepository.cs
using MotorcycleContest.Domain;

namespace MotorcycleContest.Repository;

public interface IUserRepository : IRepository<Guid, User>
{
    /**
     * Find a user by its username and password.
     * @param username the username of the user to find.
     * @param password the password of the user to find.
     * @return the user with the given username and password or null if there is no user with the given username and password.
     */
    User? FindUserByCredentials(string username, string password);
}